src/components/LmsConfigurations/DegreedIntegrationConfigForm.test.jsx

```diff
@@ -0,0 +1,67 @@
+import {
+  render, fireEvent, screen, act,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import React from 'react';
+import DegreedIntegrationConfigForm from './DegreedIntegrationConfigForm';
+import LmsApiService from '../../data/services/LmsApiService';
+
+jest.mock('../../data/services/LmsApiService', () => ({
+  postNewDegreedConfig: jest.fn(),
+  updateDegreedConfig: jest.fn(),
+}));
+
+const enterpriseId = 'test-enterprise';
+
+describe('<DegreedIntegrationConfigForm />', () => {
+  test('renders Degreed Config Form', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    // Verify all expected fields are present.
+    screen.getByLabelText('Active');
+    screen.getByLabelText('Degreed User ID');
+    screen.getByLabelText('Degreed User Password');
+    screen.getByLabelText('Degreed Instance URL');
+    screen.getByLabelText('Degreed Organization Code');
+    screen.getByLabelText('API Client ID');
+    screen.getByLabelText('API Client Secret');
+  });
+
+  test('required fields show as invalid when not filled in', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    fireEvent.click(screen.getByText('Submit'));
+
+    expect(screen.getByLabelText('Degreed User ID')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed User Password')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed Instance URL')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed Organization Code')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('API Client ID')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('API Client Secret')).toHaveClass('is-invalid');
+  });
+
+  test('submit calls postNewDegreedConfig when config is not present', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    act(() => {
+      fireEvent.change(screen.getByLabelText('Degreed User ID'), {
+        target: { value: 'testing' },
+      });
+      fireEvent.change(screen.getByLabelText('Degreed User Password'), {
+        target: { value: 'testing' },
+      });
+      fireEvent.change(screen.getByLabelText('Degreed Instance URL'), {
+        target: { value: 'testing' },
+      });
+      fireEvent.change(screen.getByLabelText('Degreed Organization Code'), {
+        target: { value: 'testing' },
+      });
+      fireEvent.change(screen.getByLabelText('API Client ID'), {
+        target: { value: 'testing' },
+      });
+      fireEvent.change(screen.getByLabelText('API Client Secret'), {
+        target: { value: 'testing' },
+      });
+      fireEvent.click(screen.getByText('Submit'));
+    });
+
+    expect(LmsApiService.postNewDegreedConfig).toHaveBeenCalled();
```

Comment:
So this test technically passes, but it is throwing a console error:
```
console.error
    Warning: An update to DegreedIntegrationConfigForm inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
        in DegreedIntegrationConfigForm
```

I _could_ ignore it, but I feel like maybe there is a better way to do this. I tried just using the regular ol' mounting and spying, but it's just fundamentally not working and I don't know if it's because of the functional component aspect or what. We don't have a lot of great examples in the codebase and functional components + testing functionality do not seem to go hand in hand on the interwebs. :(

Reply:
Oh this has been PLAUGING me lately. Wrapping it in `act` doesn't work. You might try making the test function async and `await`ing the fireEvent, but fireEvent is supposedly wrapped in act anyway. I've got other tests with this issue; I think let's ask about it on fedex and see if anyone has a good solution and if not, leave it.

Reply:
Hmm yeah I'll try the await/async in the meantime. I tried to wrap every fireEvent in act lmao and it still didn't work.